PROTO_tests/tests/setops_test.py

```diff
@@ -0,0 +1,783 @@
+    @pytest.mark.parametrize("dtype2", INTEGRAL_TYPES)
+    def test_in1d_multiarray_cross_type(self, dtype1, dtype2):
+        a, c = self.make_np_arrays_cross_type(dtype1)
+        b, d = self.make_np_arrays_cross_type(dtype2)
+
+        l1 = [ak.array(a), ak.array(b)]
+        l2 = [ak.array(c), ak.array(d)]
+        ak_result = ak.in1d(l1, l2)
+
+        la = [(x, y) for x, y in zip(a, b)]
+        lb = [(x, y) for x, y in zip(c, d)]
+        lr = [x in lb for x in la]
+
+        assert ak_result.to_list() == lr
+
+    @pytest.mark.parametrize("size", pytest.prob_size)
+    def test_in1d_multiarray_str(self, size):
+        # large general test
+        a = ak.random_strings_uniform(1, 5, size)
+        b = ak.random_strings_uniform(1, 5, size)
+
+        c = ak.random_strings_uniform(1, 5, size)
+        d = ak.random_strings_uniform(1, 5, size)
+
+        l1 = [a, b]
+        l2 = [c, d]
+
+        ak_result = ak.in1d(l1, l2)
+
+        la = [(x, y) for x, y in zip(a.to_list(), b.to_list())]
+        lb = [(x, y) for x, y in zip(c.to_list(), d.to_list())]
+        lr = [x in lb for x in la]
+        assert ak_result.to_list() == lr
+
+        stringsOne = ak.array(["String {}".format(i % 3) for i in range(10)])
+        stringsTwo = ak.array(["String {}".format(i % 2) for i in range(10)])
+        assert [(x % 3) < 2 for x in range(10)] == ak.in1d(stringsOne, stringsTwo).to_list()
+
+    @pytest.mark.parametrize("size", pytest.prob_size)
+    def test_in1d_multiarray_categorical(self, size):
+        # large general test
+        a = ak.Categorical(ak.random_strings_uniform(1, 5, size))
+        b = ak.Categorical(ak.random_strings_uniform(1, 5, size))
+
+        c = ak.Categorical(ak.random_strings_uniform(1, 5, size))
+        d = ak.Categorical(ak.random_strings_uniform(1, 5, size))
+
+        l1 = [a, b]
+        l2 = [c, d]
+
+        ak_result = ak.in1d(l1, l2)
+
+        la = [(x, y) for x, y in zip(a.to_list(), b.to_list())]
+        lb = [(x, y) for x, y in zip(c.to_list(), d.to_list())]
+        lr = [x in lb for x in la]
+        assert ak_result.to_list() == lr
+
+        stringsOne = ak.Categorical(ak.array(["String {}".format(i % 3) for i in range(10)]))
+        stringsTwo = ak.Categorical(ak.array(["String {}".format(i % 2) for i in range(10)]))
+        assert [(x % 3) < 2 for x in range(10)] == ak.in1d(stringsOne, stringsTwo).to_list()
+
+    @pytest.mark.parametrize("size", pytest.prob_size)
+    @pytest.mark.parametrize("dtype", INTEGRAL_TYPES)
+    def test_intersect1d_multiarray_numeric_types(self, size, dtype):
+        a, b = self.make_np_arrays(size, dtype)
+        c, d = self.make_np_arrays(size, dtype)
+
+        l1 = [ak.array(a), ak.array(c)]
+        l2 = [ak.array(b), ak.array(d)]
+
+        ak_result = ak.intersect1d(l1, l2)
+
+        la = set([(x, y) for x, y in zip(a, c)])
+        lb = set([(x, y) for x, y in zip(b, d)])
+        lr = list(sorted(la.intersection(lb)))
```

Comment:
This equivalent to just passing the `zip` directly into `set`.  Also `sorted` returns a list so there's no need to pass it into `list`. These same comments apply in several spots below this
```suggestion
        la = set(zip(a, c))
        lb = set(zip(b, d))
        lr = sorted(la.intersection(lb))
```

Reply:
updated this as well. I think I got everything, but if you could make sure I did not miss any, that would be great.